### PR TITLE
Set more build-flags in `setup.py`s

### PR DESCRIPTION
**Pull request by soltanmm**
Removes MinGW flags from build_python.sh script, since they're now set in our `setup.py`s by default. Also removes the arguments for the extensions in grpcio's setup.py; that function was already accessing globals anyway, might as well go whole-hog.

Needs some portability tests ran on Jenkins.

### CODE REVIEW

**Comment by soltanmm** [comment]
And by portability, I mean the artifact build portability. Not the portability tests that we run on Jenkins. Nomenclature is hard.

**Comment by soltanmm** [comment]
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/1226/

**Comment by soltanmm** [comment]
Hm, looks like a `GRPC_PYTHON_BUILD_WITH_CYTHON` flag thing snuck into this PR. Feel free to demand it be in a different commit or PR.

File: tools/distrib/python/grpcio_tools/setup.py [hunk 1]
```diff
@@ -51,15 +51,39 @@
...
+  EXTRA_ENV_COMPILE_ARGS = '-fno-wrapv -frtti -std=c++11'
+  if 'win32' in sys.platform:
+    # We use define flags here and don't directly add to DEFINE_MACROS below to
+    # ensure that the expert user/builder has a way of turning it off (via the
+    # envvars) without adding yet more GRPC-specific envvars.
+    EXTRA_ENV_COMPILE_ARGS += ' -D_ftime=_ftime64 -D_timeb=__timeb64'
+if EXTRA_ENV_LINK_ARGS is None:
+  EXTRA_ENV_LINK_ARGS = '-lpthread'
+  if 'win32' in sys.platform:
+    # TODO(atash) check if this is actually safe to just import and call on
```
Comment (kpayson64) [inline tools/distrib/python/grpcio_tools/setup.py]: Verified that that distutils.cygwinccompiler is a pure python module that exists on all distributions.

```get_msvcr()``` returns None on Linux, but I think it makes sense to only call ```get_msvcr()``` within an ```if 'win32' in sys.platform``` block

File: tools/distrib/python/grpcio_tools/setup.py [hunk 2]
```diff
@@ -51,15 +51,39 @@
...
+    # non-Windows (to avoid breaking import style)
+    from distutils.cygwinccompiler import get_msvcr; print(get_msvcr()[0])
```
Comment (kpayson64) [inline tools/distrib/python/grpcio_tools/setup.py]: We can remove the print now :)

File: tools/distrib/python/grpcio_tools/setup.py [hunk 3]
```diff
@@ -51,15 +51,39 @@
...
+    msvcr = get_msvcr()[0]
+    EXTRA_ENV_LINK_ARGS += (
+        ' -static-libgcc -static-libstdc++ -mcrtdll={msvcr} '
+        '-static'.format(msvcr))
```
Comment (kpayson64) [inline tools/distrib/python/grpcio_tools/setup.py]: I think when you have a ```{keyword}``` format block, you need to do ```.format(keyword=keyword)```

File: tools/distrib/python/grpcio_tools/setup.py (hunk 2 above)
Comment (soltanmm) [inline tools/distrib/python/grpcio_tools/setup.py]: blargjhaflarghin yep.

File: tools/distrib/python/grpcio_tools/setup.py (hunk 3 above)
Comment (soltanmm) [inline tools/distrib/python/grpcio_tools/setup.py]: Now that you mention it, I could've sworn I believed that to be true. Seeing it here and not seeing a failing test is confusing. Why didn't it fail in the build? Hm...

**Comment by soltanmm** [comment]
Running through with another artifact build... The previous one may be stale.

**Comment by soltanmm** [comment]
https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/1237/

**Comment by soltanmm** [comment]
Kicking portability tests...

**Comment by kpayson64** [comment]
LGTM

**Pull request merged**